server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -438,6 +439,46 @@ public Map<String, NamedAnalyzer> indexAnalyzers() {
         return Map.of();
     }
 
+    /**
+     * Specifies the mode of synthetic source support by the mapper.
+     *
+     * <pre>
+     * {@link NATIVE} - mapper natively supports synthetic source, f.e. by constructing it from doc values.
+     *
+     * {@link FALLBACK} - mapper does not have native support but uses fallback implementation.
+     *
+     * {@link NOT_SUPPORTED} - synthetic source is not supported.
+     * </pre>
+     */
+    protected enum SyntheticSourceMode {
+        NATIVE,
+        FALLBACK,
+        NOT_SUPPORTED
+    }
+
+    /**
+     * Specifies the mode of synthetic source support by the mapper.
+     *
+     * @return {@link SyntheticSourceMode}
+     */
+    protected SyntheticSourceMode syntheticSourceMode() {
```

Comment:
This method will remain but maybe `FALLBACK` variant will be removed once we apply fallback everywhere. Just for context - we need this separate piece of information because `syntheticFieldLoader` throws exceptions and we can't call it to determine if mapper supports synthetic source natively. If we do that, that exception will be wrapped in document parsing exception which is different from how it's currently thrown.